src/utils/records.js

```diff
@@ -1,6 +1,20 @@
 import { addressUtils } from '@0xproject/utils'
 import { validateContent } from './contents'
+
+/**
+ * @typedef { "address" | "content" | "oldcontent" } recordType
+ */
+
+/**
+ * Validate Record
+ * @param {Object} record
+ * @param {recordType} record.type The record type
+ * @param {string} record.value The record value
+ * @throws {Error} will throw an error if unrecognised record type
+ * @returns {boolean|any}
+ */
 export function validateRecord(record) {
+  // todo removeme Info: using typescript means that we can verify record.type whilst we are coding/compile time instead of at run time
```

Comment:
You may not need it for typescript but still needed as a javascript library. If you want to convert the entire library to typescript, feel free :-)

Reply:
I wish I could have enough hours in the day to do that haha!

Reply:
Yes you are right, still needed for JS library (I shouldn't have used the word 'instead')